src/webapp/src/filters/http-exceptions.filter.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpException,
+  UnauthorizedException,
+  ForbiddenException,
+  NotFoundException,
+  InternalServerErrorException
+} from '@nestjs/common'
+import { Request, Response } from 'express'
+
+@Catch()
+export class HttpExceptionsFilter implements ExceptionFilter {
+  catch (exception: Error, host: ArgumentsHost): any {
+    const ctx = host.switchToHttp()
+    const response = ctx.getResponse<Response>()
+    const request: any = ctx.getRequest<Request>()
+
+    console.log('ex', exception)
+
+    // This needs to be first because it is a special
+    // kind of unauthrized that does not trigger any
+    // logout and needs to return a 302
+    if (
+      exception instanceof UnauthorizedException &&
+      exception.message === 'Subscription not active'
+    ) {
+      if (request.is('json') === true || request?.query?.json != null) {
+        return response.json({
+          statusCode: 302,
+          message: 'Found',
+          redirect: '/user/billing'
+        })
+      }
+
+      return response.redirect('/user/billing')
+      // TODO
+      // response.redirect('/user');
+    }
+
+    const e = exception instanceof HttpException ? exception : new InternalServerErrorException()
+    if (request?.query?.json != null) {
+      return response
+        .status(e.getStatus())
+        .json(e.getResponse())
+    }
+
+    const themeRoot = 'default'
+    const assetsRoot = `/${themeRoot}`
+
+    if (
+      // view doesn't exist for tenant, revert to default
+      exception.message.startsWith('Failed to lookup view')
+    ) {
+      const assetsRoot = '/default'
+      response.statusCode = 500
+      response.render(`${themeRoot}/500`, { assetsRoot })
+    } else if (
+      exception instanceof UnauthorizedException ||
+      exception instanceof ForbiddenException
+    ) {
+      let next: string = ''
+      const originalUrl = request.originalUrl
+
+      switch (exception.message) {
+        case 'admin':
+          response.redirect('/')
+          break
+        case 'Invalid email or password':
+          response.render(`${themeRoot}/error`, { error: 'Invalid email or password' })
```

Comment:
I don't like that we are directly rendering a page here:
- it feels the wrong place
- The amount of data we can send to the page is very limited (e.g. we don't easily have the setting
-  I'm not 100% sure we should handle wrong username or password with exceptions